### PR TITLE
MASE

### PR DESCRIPTION
**Pull request by j1c**
#### Reference Issues/PRs
Implements joint RDPG, now called MASE.

#### Any other comments?
Still very unclear how to specify the model when the underlying dimensions is unknown due to the two step embedding process. We could allow for different `n_components` and/or `n_elbows` for each step, but no one really knows how we should parameterize this.

Other changes:
- modified `import_graph` so that it actually works on tensor inputs
- added `BaseEmbedMulti` that abstracts input checking function.
- updated omni accordingly

### CODE REVIEW

**Comment by bdpedigo** [comment]
@j1c Do you have a link to the paper/manuscript?

File: graspy/utils/utils.py [hunk 1]
```diff
@@ -38,14 +38,23 @@ def import_graph(graph):
     if isinstance(graph, (nx.Graph, nx.DiGraph, nx.MultiGraph, nx.MultiDiGraph)):
         out = nx.to_numpy_array(graph, nodelist=sorted(graph.nodes), dtype=np.float)
     elif isinstance(graph, (np.ndarray, np.memmap)):
-        maximum = np.max(graph.shape)
+        shape = graph.shape
+        if len(shape) == 3:
```
Comment (bdpedigo) [inline graspy/utils/utils.py]: is there any check for len(shape) > 4? If not I think line 49 could break if there is a 4+d array and one of the dimensions is really long

File: graspy/embed/jrdpg.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
+import warnings
```
Comment (bdpedigo) [inline graspy/embed/jrdpg.py]: add new header

File: graspy/embed/jrdpg.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+        # Check if undirected
+        undirected = all(is_almost_symmetric(g) for g in graphs)
+
+        # embed
+        Uhat, Vhat = self._reduce_dim(graphs)
+        self.latent_left_ = Uhat
+        if not undirected:
+            self.latent_right_ = Vhat
+            self.scores_ = Uhat.T @ graphs @ Vhat
```
[the remark this answers is not in the document]
Comment (j1c) [reply graspy/embed/jrdpg.py]: Not really. This isn't a classifier, so there shouldn't be any confusion between a method that doesn't exist in this class vs an attribute.

If you can think of a better name, Im open, but I am not going to call it `Rhats`.

**Comment by j1c** [comment]
Ah fuck i messed up with history

**Pull request closed without merging**